### PR TITLE
Hotfix: DatacatsError for undocumented exceptions failing in formatting

### PR DESCRIPTION
**Pull request by dborzov**
Fixes [issue#190](https://github.com/datacats/datacats/issues/190)

`.format(args)` should be called only once on `DatcatsExeptions.message`. It was not.

### CODE REVIEW

**Comment by JackMc** [comment]
Pls fix CI

**Comment by dborzov** [comment]
@JackMc right, thanks, Jack. Done

File: datacats/error.py [hunk 1]
```diff
@@ -4,7 +4,10 @@
 class DatacatsError(Exception):
 
     def __init__(self, message, format_args=(), parent_exception=None):
-        self.message = message
+        try:
+            self.message = message.format(format_args)
+        except (ValueError, KeyError):
+            self.message = message
```
Comment (wardi) [inline datacats/error.py]: this is odd. Why would you be getting Value/KeyErrors it's only our code that creates these.

Comment (wardi) [inline datacats/error.py]: And shouldn't this be `message.format(*format_args)`

Comment (dborzov) [inline datacats/error.py]: You are right, it should be `*format_args`, thanks.

ValueError and KeyError are raised for this two cases correspondingly:

```python
"I like {".format(fruit=bananas) # ValueError
"I like {vegetable} ".format(fruit=bananas) # KeyError
```

File: datacats/error.py [hunk 2]
```diff
@@ -4,7 +4,10 @@
 class DatacatsError(Exception):
 
     def __init__(self, message, format_args=(), parent_exception=None):
-        self.message = message
+        try:
+            self.message = message.format(*format_args)
+        except (ValueError, KeyError):
```
Comment (dborzov) [inline datacats/error.py]: You are right, it should be *format_args, thanks.

ValueError and KeyError are raised for these two cases correspondingly:

"I like {".format(fruit=bananas) # ValueError
"I like {vegetable} ".format(fruit=bananas) # KeyError

Comment (wardi) [inline datacats/error.py]: Both those cases would be our code with an error that shouldn't be silently ignored.

Better yet: let's remove format_args from the constructor and let the code that creates these instances call .format() if it wants to on the message it passes in. That way the error stays right where it was caused and unrelated code stays out of our exception classes.

Comment (dborzov) [inline datacats/error.py]: I agree. This was the original format for DatacatsError and it is really obscure and error prone. Not only parameters are passed to the exception, they are not named. So you have to count "{0}, {1}" each time and it is really obscure.

**Comment by wardi** [comment]
merging, I'll make a new issue for moving string formatting out of the DatacatsError class
